packages/itwin/ec3-widget/src/test/ExportModal.test.tsx

```diff
@@ -0,0 +1,151 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import React from "react";
+import "@testing-library/jest-dom";
+import { act, screen } from "@testing-library/react";
+import * as moq from "typemoq";
+import type { EC3Job, IEC3JobsClient } from "@itwin/insights-client";
+import { CarbonUploadState } from "@itwin/insights-client";
+import faker from "@faker-js/faker";
+import { renderWithContext } from "./test-utils";
+import { ExportModal } from "../components/ExportModal";
+
+jest.mock("@itwin/itwinui-react", () => ({
+  ...jest.requireActual("@itwin/itwinui-react"),
+  toaster: {
+    positive: (_: string) => { },
+    negative: (_: string) => { },
+  },
+}));
+
+const jobsClient = moq.Mock.ofType<IEC3JobsClient>();
+
+describe("ExportModal", () => {
+  const templateId = faker.datatype.uuid();
+  const jobId = faker.datatype.uuid();
+
+  const job: EC3Job = {
+    id: jobId,
+    _links: {
+      status: {
+        href: "status",
+      },
+    },
+  };
+
+  function status(state: CarbonUploadState) {
+    return {
+      status: state,
+      _links: {
+        ec3Project: {
+          href: jobId,
+        },
+      },
+    };
+  }
+
+  const accessToken = faker.datatype.uuid();
+  const ec3Token = faker.datatype.uuid();
+  const getAccessTokenFn = async () => accessToken;
+
+  beforeAll(async () => {
+    jobsClient.setup(async (x) => x.createJob(accessToken, moq.It.isAny())).returns(async () => job);
+  });
+
+  it("Export modal with the isOpen prop should render successfully and be visible", async () => {
+    await renderWithContext({
+      component: <ExportModal
+        projectName=""
+        isOpen={true}
+        close={() => { }}
+        templateId={templateId}
+        token={undefined}
+      />,
+    });
+    expect(screen.getByTestId("ec3-export-modal")).toBeDefined();
+    expect(document.querySelectorAll(".iui-dialog-visible")).toBeDefined();
+  });
+
+  it("Export modal without the isOpen prop should be invisible", async () => {
+    await renderWithContext({
+      component: <ExportModal
+        projectName=""
+        isOpen={false}
+        close={() => { }}
+        templateId={templateId}
+        token={undefined}
+      />,
+    });
+    expect(document.querySelector(".ec3-export-modal")).toBeDefined();
+    expect(document.querySelector(".iui-dialog-visible")).toBeNull();
+  });
+
+  it("Interval should be set when modal is open and ec3 token is received", async () => {
+    let event: Function | undefined;
+    jest.spyOn(window, "setInterval").mockImplementation((callback, _) => {
```

Comment:
There is nothing that resets this spy / mock. Might introduce issues later on. Better reset all the mocks after tests (`afterEach`, `afterAll`)

Reply:
added an afterEach function to all tests that restores all mocks This does not remove the mocks of clients since another library is used for those mocks.